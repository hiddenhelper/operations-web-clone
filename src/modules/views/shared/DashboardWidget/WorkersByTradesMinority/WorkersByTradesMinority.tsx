import React, { memo, useMemo, useEffect } from 'react';

import { GeneralModel, StatisticsModel } from '../../../../models';
import DashboardTopTen from '../DashboardTopTen';

export interface IWorkersByTradesMinorityProps {
  queryParams: GeneralModel.IQueryParams;
  topTenStatisticsMap: GeneralModel.IEntityMap<StatisticsModel.ITopTenStatistics[]>;
  loading: GeneralModel.ILoadingStatus;
  fetchWorkersByTradesMinority: (key: string, query: GeneralModel.IQueryParams) => void;
}

const WorkersByTradesMinority = ({ queryParams, topTenStatisticsMap, loading, fetchWorkersByTradesMinority }: IWorkersByTradesMinorityProps) => {
  const idKey = useMemo(() => 'workersByTradesMinority', []);
  const workersByTradesMinorityList = useMemo(() => topTenStatisticsMap[idKey] || [], [topTenStatisticsMap, idKey]);
  const isLoading: boolean = useMemo(() => loading && loading.isLoading, [loading]);

  useEffect(() => {
    fetchWorkersByTradesMinority(idKey, queryParams);
  }, [queryParams, idKey, fetchWorkersByTradesMinority]);
  return (
    <DashboardTopTen
      title="Workers by Trades - Minority"
      itemTitle="Trade"
      isLoading={isLoading}
      list={workersByTradesMinorityList}
      styles={{ minHeight: '100%' }}
    />
  );
};

export default memo(WorkersByTradesMinority);
