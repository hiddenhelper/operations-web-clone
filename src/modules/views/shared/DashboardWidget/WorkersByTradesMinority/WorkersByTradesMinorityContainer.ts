import { Dispatch } from 'redux';
import { connect } from 'react-redux';

import { GeneralModel } from '../../../../models';
import { GENERAL } from '../../../../../constants';
import { IRootState } from '../../../../state-mgmt/rootState';
import { statisticsState } from '../../../../state-mgmt/statistics';

import WorkersByTradesMinority from './WorkersByTradesMinority';

export const mapStateToProps = (state: IRootState) => ({
  topTenStatisticsMap: state.statistics.topTenStatisticsMap,
  loading: state.general.loadingMap[GENERAL.LOADING_KEY.FETCH_WORKERS_BY_TRADES_MINORITY],
});

export const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchWorkersByTradesMinority: (key: string, query: GeneralModel.IQueryParams) =>
    dispatch(statisticsState.actions.fetchWorkersByTradesMinorityStart(key, query)),
});

export default connect(mapStateToProps, mapDispatchToProps)(WorkersByTradesMinority);
