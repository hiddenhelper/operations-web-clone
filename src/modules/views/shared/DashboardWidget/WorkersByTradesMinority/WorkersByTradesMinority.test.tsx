import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import { getBadgeByLocationWidget_1 } from '../../../../../test/entities';
import WorkersByTradesMinority, { IWorkersByTradesMinorityProps } from './WorkersByTradesMinority';

describe('WorkersByTradesMinority Component', () => {
  global.console.warn = () => {
    /** */
  };
  let wrapper: RenderResult;
  let props: IWorkersByTradesMinorityProps;

  beforeEach(() => {
    const mockDate = jest.fn(() => new Date(Date.UTC(2020, 11, 14)).valueOf());
    Date.now = mockDate;
    props = {
      queryParams: { stateCode: '', period: 0 },
      topTenStatisticsMap: { workersByTradesMinority: getBadgeByLocationWidget_1() },
      loading: {
        isLoading: false,
        hasError: false,
        error: null,
      },
      fetchWorkersByTradesMinority: jest.fn(),
    };
  });

  it('should render', () => {
    wrapper = render(
      <MemoryRouter>
        <WorkersByTradesMinority {...props} />
      </MemoryRouter>
    );
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should fetchWorkersByTradesMinority', () => {
    props.topTenStatisticsMap = {};
    render(
      <MemoryRouter>
        <WorkersByTradesMinority {...props} />
      </MemoryRouter>
    );
    expect(props.fetchWorkersByTradesMinority).toHaveBeenCalled();
  });

  it('should render loading', () => {
    props.loading = { isLoading: true, hasError: false, error: null };
    wrapper = render(
      <MemoryRouter>
        <WorkersByTradesMinority {...props} />
      </MemoryRouter>
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
