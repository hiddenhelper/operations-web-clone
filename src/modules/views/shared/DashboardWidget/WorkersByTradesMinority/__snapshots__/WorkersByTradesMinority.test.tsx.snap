// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkersByTradesMinority Component should render 1`] = `
<div>
  <div
    class="MuiPaper-root  makeStyles-widget-57 MuiPaper-elevation0 MuiPaper-rounded"
    style="height: 100%; min-height: 100%;"
  >
    <div
      class="makeStyles-widgetTitleMarginBottom-40 makeStyles-withFilter-55"
    >
      <p
        class="MuiTypography-root makeStyles-widgetTitle-59 makeStyles-widgetTitleInlinePosition-74 MuiTypography-body1"
      >
        Workers by Trades - Minority
      </p>
    </div>
    <div
      style="padding-top: 15px;"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 12px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
              style="display: flex; align-items: center;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
                style="font-weight: bold;"
              >
                Trade
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
                style="font-weight: bold;"
              >
                Quantity
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="Huston"
              >
                Huston
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 85%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              938
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="Dallas"
              >
                Dallas
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 80%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              839
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="Austin"
              >
                Austin
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 78%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              783
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="San Antonio"
              >
                San Antonio
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 73%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              762
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="El Paso"
              >
                El Paso
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 65%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              684
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="Leage City"
              >
                Leage City
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 60%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              631
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="For Worth"
              >
                For Worth
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 50%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              603
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="Texas City"
              >
                Texas City
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 46%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              589
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="Amarillo"
              >
                Amarillo
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 41%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              524
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 6.5px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
            >
              <div
                class=""
                style="align-items: center; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
                title="Waco"
              >
                Waco
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8 MuiGrid-grid-xl-8"
              style="display: flex; align-items: center; padding-left: 5px;"
            >
              <div
                style="width: 35%; background-color: rgb(233, 166, 31); height: 8px; border-radius: 4px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-lg-1 MuiGrid-grid-xl-1"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              389
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkersByTradesMinority Component should render loading 1`] = `
<div>
  <div
    class="MuiPaper-root  makeStyles-widget-242 MuiPaper-elevation0 MuiPaper-rounded"
    style="height: 100%; min-height: 100%;"
  >
    <div
      class="makeStyles-widgetTitleMarginBottom-225 makeStyles-withFilter-240"
    >
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: 30px;"
      />
    </div>
    <div
      style="padding-top: 15px;"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
        style="height: 24px; margin-bottom: 12px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
              style="display: flex; align-items: center;"
            >
              <div
                class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
                style="width: 30%; height: 15px;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
              style="display: flex; align-items: center; justify-content: flex-end;"
            >
              <div
                class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
                style="width: 30%; height: 15px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
        style="width: 100%; height: auto;"
      />
    </div>
  </div>
</div>
`;
